dips/dip-12.md

```diff
@@ -0,0 +1,56 @@
+---
+dip: 12
+title: Blockchain Coin Trade Flow
+authors: Lu Zhang (@LuZhang-Lou)
+dip manager: David Wolinsky (@davidiw)
+status: Draft
+type: Informational
+created: 3/5/2021
+---
+# Summary and Motivation
+This document proposes a standardized method for coin trade flows in the Diem Network. The flow is generic for coin purchase and coin sell transactions. It supports a single trade settlement and batch settlement.
+
+
+A Blockchain coin trade is the process where an entity, for example a Virtual Asset Service Provider (VASP), transacts with a DD to purchase or sell coins. Beyond on-chain p2p transactions, a coin trade usually involves several stages, such as
+  1. trade initialization
+  2. fiat currency (or other types of cryptocurrency) debitted
+  3. on-chain transfer.
+ The first couple of steps are typically unique to the parties involved. However, we see the possibilities and benefits to generalize the on-chain phase, in pursuit of interoperability among Diem Blockchain participants.
+
+# Terminology
+* trade_id - a unified trade identifier used over the course, a UTF-8 string. The recommended max length is 255 bytes. It is agreed upon, by both entities, during the trade initialization and negotiation phase. It should be unique for every trade between a buyer and a seller.
```

Comment:
Why is a trade_id expected to be a UTF-8 string rather than a u64/u128 for instance?

Reply:
good call, best to say that at trade_id is an identifier defined by the two parties involved and is therefore represented by a UTF-8 string with a recommended maximum length of 255 bytes (or characters?).

Reply:
@aching we've seen non-numeric trade id from some partner. Usually this id is the one that is included in the trade confirmation docs, so if we make them numeric then it adds another layer of transformation for them.